test/server/http_time_tracking_filter_integration_test.cc

```diff
@@ -0,0 +1,147 @@
+#include <chrono>
+
+#include "envoy/upstream/cluster_manager.h"
+#include "envoy/upstream/upstream.h"
+
+#include "external/envoy/test/common/upstream/utility.h"
+#include "external/envoy/test/integration/http_integration.h"
+#include "external/envoy/test/test_common/simulated_time_system.h"
+
+#include "api/server/response_options.pb.h"
+#include "api/server/response_options.pb.validate.h"
+
+#include "server/configuration.h"
+#include "server/http_time_tracking_filter.h"
+#include "server/well_known_headers.h"
+
+#include "gtest/gtest.h"
+
+namespace Nighthawk {
+
+using namespace std::chrono_literals;
+
+const std::string kLatencyResponseHeaderName = "x-prd";
+const std::string kDefaultProtoFragment = fmt::format(
+    "emit_previous_request_delta_in_response_header: \"{}\"", kLatencyResponseHeaderName);
+const std::string kProtoConfigTemplate = R"EOF(
+name: time-tracking
+typed_config:
+  "@type": type.googleapis.com/nighthawk.server.ResponseOptions
+  {}
+)EOF";
+
+class HttpTimeTrackingIntegrationTest
```

Comment:
In the other test file, we define classes, then the tests they're for, which seems a little cleaner to me than defining all of the classes, then all of the tests.  Is it possible to do that here too?